File: apk/app/src/main/java/com/example/momentra/ui/setup/SetupPreferenceFilter.kt
package com.example.momentra.ui.setup

/** Filters wizard state to keys accepted by backend catalog defaults. */
object SetupPreferenceFilter {
    fun filterToCatalogKeys(
        preferences: Map<String, Any>,
        allowedKeys: Set<String>,
    ): Map<String, Any> = preferences.filterKeys { it in allowedKeys }

    fun filterLocalOnly(
        preferences: Map<String, Any>,
        localOnlyKeys: Set<String>,
    ): Map<String, Any> = preferences.filterKeys { it !in localOnlyKeys }
}

File: momentra/momentra/Shell/Setup/SetupPreferenceFilter.swift
import Foundation

/// Filters wizard state to keys accepted by backend catalog defaults.
enum SetupPreferenceFilter {
    static func filterToCatalogKeys(
        _ preferences: [String: Any],
        allowedKeys: Set<String>
    ) -> [String: Any] {
        preferences.filter { allowedKeys.contains($0.key) }
    }

    static func filterLocalOnly(
        _ preferences: [String: Any],
        localOnlyKeys: Set<String>
    ) -> [String: Any] {
        preferences.filter { !localOnlyKeys.contains($0.key) }
    }
}
